pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -180,22 +208,45 @@ func (r *ReconcileConstraintTemplate) Reconcile(request reconcile.Request) (reco
 		found := &apiextensionsv1beta1.CustomResourceDefinition{}
 		err = r.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, found)
 		if err != nil && errors.IsNotFound(err) {
-			return r.handleCreate(instance, crd)
+			result, err := r.handleCreate(instance, crd)
+			if err != nil {
+				reportError()
+			}
+			if !result.Requeue {
+				reportSuccess()
+			}
```

Comment:
If err is nil and result.Requeue is true, then the create will not get recorded. Same below.

Reply:
This is intentional, as a requeue means that the constraint is not fully reconciled (examples include problems setting the finalizer (L211 in original code) or problems reporting status (L310 in original code). Because it's a requeue the metric is expected to get updated on the next iteration.

What's weird here is that because a requeue means that we are in an inconsistent state, it's hard to know when it's most appropriate to increment the metric. I chose to defer incrementing as long as possible to make it easier to spot problems by comparing the # provided by the metric against the # of constraints on the API server. If there is a consistent shortfall, some object is not reconciling correct.